Source/Core/Xml.cs

```diff
@@ -109,6 +109,41 @@ public void WriteEndMethod(string outcome, DateTime endTime, TimeSpan elapsed)
       cce.EndExpose();
     }
 
+    public void WriteStartSplit(int splitNum, DateTime startTime)
+    {
+      Contract.Requires(splitNum > 0);
```

Comment:
Are these contracts really useful? I thought everything but `Contract.Assert` has no runtime or compile time effect, and I would argue even the `Contract.Assert` is more noisy than useful.

Reply:
Yup, I was just following the established pattern for now. I do find they are useful documentation even if they are never enforced.

As an aside, I was pleasantly surprised that the build rejected my attempt to add null checks that were unnecessary, but I'm now confused as to how the project is using the new nullability rules since I can't find any `<Nullable>` elements in the configuration.